sdk/eventhub/event-hubs/src/eventProcessor.ts

```diff
@@ -513,7 +537,9 @@ export class EventProcessor {
 
       // waits for the event processor loop to complete
       // will complete immediately if _loopTask is undefined
-      await this._loopTask;
+      if (this._loopTask) {
```

Comment:
Should we also only abandon partitions if we are processing multiple partitions? (Line 549, but I can't leave a comment there)

Reply:
Great point, might as well avoid that call too. Gave me a chance to make the "target type" check a bit nicer too, so good improvement over all.